cmd/rbac-manager/main.go

```diff
@@ -110,6 +112,30 @@ func createGoogleDirectory(ctx context.Context, subject string) (*directoryv1.Se
 		return nil, err
 	}
 
+	// If the found credential doesn't contain JSON, try to fallback to workload identity
+	if len(creds.JSON) == 0 {
+		// Get the email address associated with the service account. The account may be empty
+		// or the string "default" to use the instance's main account.
+		principal, err := metadata.Email("default")
+		if err != nil {
+			return nil, err
+		}
+
+		// Access to the directory API must be signed with a Subject to enable domain selection.
+		config := impersonate.CredentialsConfig{
+			TargetPrincipal: principal,
+			Scopes:          scopes,
+			Subject:         subject,
+		}
+
+		ts, err := impersonate.CredentialsTokenSource(ctx, config, option.WithCredentials(creds))
+		if err != nil {
+			return nil, err
+		}
+
+		return directoryv1.NewService(ctx, option.WithTokenSource(ts))
```

Comment:
I wonder if the option when using a supplied json credentials (old behavior) could also use the TokenSource from `Credentials` returned by `FindDefaultCredentials`?

If so, likely the code could be simplified a little, instead of needing to call `NewService` with different option for each credential type.

Reply:
Yeap that's a good point! I didn't change that initially as I didn't want to make too much changes to existing code.

But you're (mostly) right that we can use the same `WithTokenSource` option for both the new and old behaviours. However, it's not `TokenSource` from the `Credentials` returned by `FindDefaultCredentials` but the `TokenSource` from the `jwt.Config`, as we need the token source to have the (domain-wide delegation) subject set.

Reply:
I've updated the PR to use WithTokenSource for both codepaths 👍🏻